python/knowhere/knowhere.i

```diff
@@ -354,4 +356,51 @@ DumpRangeResultDis(knowhere::DataSetPtr result, float* dis, int len) {
     }
 }
 
+void
+Dump(knowhere::BinarySetPtr binset, const std::string& file_name) {
+    auto binary_set = *binset;
+    auto binary_map = binset -> binary_map_;
+    std::ofstream outfile;
+    outfile.open(file_name, std::ios::out | std::ios::trunc);
+    for (auto it = binary_map.begin(); it != binary_map.end(); ++it) {
+        // serialization: name_length(size_t); name(char[]); binset_size(size_t); binset(uint8[]); 
+        auto name = it->first;
+        outfile << name.size();
+        outfile << name;
+        auto value = it->second;
+        outfile << value->size;
+        outfile.write(reinterpret_cast<char*>(value->data.get()), value->size);
+    }
+    // end with 0
+    outfile << 0;
+    outfile.close();
+}
+
+void
+Load(knowhere::BinarySetPtr binset, const std::string& file_name) {
+    std::ifstream infile;
+    infile.open(file_name, std::ios::in);
+    size_t name_len;
+    char flag;
+    while (true) {
+        // serialization: name_length(size_t); name(char[]); binset_size(size_t); binset(uint8[]); 
+        infile >> name_len;
+        if (name_len == 0) break;
+
+        auto _name = new char[name_len];
+        infile.read(_name, name_len);
+        std::string name(_name, name_len);
+
+        int64_t size;
+        infile >> size;
+
+        auto data = new uint8_t[size];
+        infile.read(reinterpret_cast<char*>(data), size);
+        std::shared_ptr<uint8_t[]> data_ptr(data);
```

Comment:
`std::shared_ptr<>` first, read data into it next

Reply:
how about this
`std::shared_ptr<uint8_t[]> data_ptr(data);`
`infile.read(reinterpret_cast<char*>(data_ptr.get()), size);`

Reply:
yes. 
Also, please note that this code needs C++17, check https://stackoverflow.com/questions/13061979/can-you-make-a-stdshared-ptr-manage-an-array-allocated-with-new-t , I don't know how this shared_ptr<uint8_t[]> will be used later.